src/main/java/ro/andreiciortea/yggdrasil/http/HttpServerVerticle.java

```diff
@@ -102,7 +102,11 @@ private Router createRouter() {
     router.delete("/artifacts/:artid").handler(templateHandler::handleDeleteInstance);
     router.put("/artifacts/updateTriples/:artid").handler(templateHandler::handleUpdateTriples);
     // invoke actions on software artifacts defined in the annotations of the corresponding template
-    router.put("/artifacts/:artid/*").handler(templateHandler::handleTemplateExtended);
+    router.put("/artifacts/:artid/*").handler(templateHandler::handleAction);
+    router.post("/artifacts/:artid/*").handler(templateHandler::handleAction);
+    router.put("/artifacts/:artid/*").handler(templateHandler::handleAction);
+    router.delete("/artifacts/:artid/*").handler(templateHandler::handleAction);
+    router.options("/artifacts/:artid/*").handler(templateHandler::handleAction);
```

Comment:
Why specify the methods here? At this point we cannot yet verify if a method is supported for a given artifact path, right? If we don't care about the HTTP method, we can directly route only based on the relative path and reduce this to one line of code.

Reply:
Totally makes sense. Didn't think of the router.route() function. Will change.